tensorflow/core/kernels/reduction_gpu_kernels.cu.h

```diff
@@ -502,16 +518,16 @@ void LaunchScalarReduction(OpKernelContext* ctx, OUT_T out, IN_T in,
             DT_INT8, TensorShape({static_cast<int64>(num_blocks * sizeof(T))}),
             &temp_storage));
 
-    BlockReduceKernel<IN_T, T*, num_threads>
-        <<<num_blocks, num_threads, 0, cu_stream>>>(
+    GPU_LAUNCH_KERNEL(BlockReduceKernel<IN_T, T*, num_threads>,
+        dim3(num_blocks), dim3(num_threads), 0, cu_stream,
             in, (T*)temp_storage.flat<int8_t>().data(), in_size, op, init);
 
     // take care that we only reduce blocks that had some valid elements in them
     // TODO(eriche): CUB currently has a bug in HeadSegmentedReduce that
     // requires it to be used with a full warp.  Can reduce 32 -> num_blocks
     // when this is fixed.
-    CleanupSegments<<<1, 32, 0, cu_stream>>>(
-        (T*)temp_storage.flat<int8_t>().data(), out, 1, 1, num_blocks, op,
+    GPU_LAUNCH_KERNEL(CleanupSegments<T*,OUT_T,Op>, dim3(1), dim3(32), 0, cu_stream,
```

Comment:
is this kernel launch needed in case of `TENSORFLOW_USE_ROCM`?

Reply:
Unclear, but I wanted to have a working, straight conversion first and worry about performance next.